backend/src/apiserver/server/run_server.go

```diff
@@ -118,23 +117,18 @@ func (s *RunServer) createRun(ctx context.Context, run *model.Run) (*model.Run,
 	if err != nil {
 		return nil, util.Wrapf(err, "Failed to create a run due to invalid experimentId and namespace combination")
 	}
+	if common.IsMultiUserMode() && namespace == "" {
+		return nil, util.NewInvalidInputError("A run cannot have an empty namespace in multi-user mode")
+	}
 	run.ExperimentId = experimentId
 	run.Namespace = namespace
 	// Check authorization
-	if common.IsMultiUserMode() {
-		if run.Namespace == "" {
-			return nil, util.NewInternalServerError(util.NewInvalidInputError("Run cannot have an empty namespace in multi-user mode"), "Failed to create a run due to empty namespace")
-		}
-		resourceAttributes := &authorizationv1.ResourceAttributes{
-			Namespace: run.Namespace,
-			Verb:      common.RbacResourceVerbCreate,
-			Group:     common.RbacPipelinesGroup,
-			Resource:  common.RbacResourceTypeRuns,
-			Name:      run.DisplayName,
-		}
-		if err := s.resourceManager.IsAuthorized(ctx, resourceAttributes); err != nil {
-			return nil, util.Wrapf(err, "Failed to create a run due to authorization error. Check if you have write permissions to namespace %s", run.Namespace)
-		}
+	resourceAttributes := &authorizationv1.ResourceAttributes{
+		Namespace: run.Namespace,
+		Verb:      common.RbacResourceVerbCreate,
+	}
```

Comment:
Again, here's a case that the old code sets 5 attributes in `ResourceAttributes`, while the new code sets only 2. Seems not right?

Reply:
See [here](https://github.com/kubeflow/pipelines/pull/9373/files#r1191530425)